packages/vite/src/node/plugins/css.ts

```diff
@@ -2374,7 +2374,15 @@ const makeModernScssWorker = (
               ? fileURLToPath(context.containingUrl)
               : options.filename
             const resolved = await internalCanonicalize(url, importer)
-            return resolved ? pathToFileURL(resolved) : null
+            if (
+              resolved &&
+              (resolved.endsWith('.css') ||
+                resolved.endsWith('.scss') ||
+                resolved.endsWith('.sass'))
```

Comment:
```suggestion
              // only limit to these extensions because:
              // - for the `@import`/`@use`s written in file loaded by `load` function,
              //   the `canonicalize` function of that `importer` is called first
              // - the `load` function of an importer is only called for the importer
              //   that returned a non-null result from its `canonicalize` function
              (resolved.endsWith('.css') ||
                resolved.endsWith('.scss') ||
                resolved.endsWith('.sass'))
```